pkg/optimizely/cache.go

```diff
@@ -103,12 +105,32 @@ func (c *OptlyCache) Wait() {
 	c.wg.Wait()
 }
 
+var errValidationFailure = errors.New("sdkKey failed validation")
+
+func regexValidator(sdkKeyRegex string) func(sdkkey string) bool {
+	r, err := regexp.Compile(sdkKeyRegex)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("invalid sdkKeyRegex configuration")
```

Comment:
This will cause Agent to `os.Exit(1)`
https://github.com/rs/zerolog/blob/9e51190d4708789500a087352d12d5f2e0c30b0d/log.go#L332-L338